src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -409,90 +409,58 @@ private static void RemoveToolLanguage(JObject run)
         {
-            // Access and modify run.tool
-            JObject tool = (JObject)run["tool"];
-            UpdateToolReportingDescriptorPropertyTypes(tool);
-
-            // Access and modify run.conversion.tool
-            if (run["conversion"] is JObject conversion)
+            string[] reportingDescriptorPathsToUpdate =
             {
-                tool = (JObject)conversion["tool"];
-                UpdateToolReportingDescriptorPropertyTypes(tool);
-            }
-        }
+                "tool.driver.notificationDescriptors[]",
+                "conversion.tool.driver.notificationDescriptors[]",
+                "tool.extensions[].notificationDescriptors[]",
+                "conversion.tool.extensions[].notificationDescriptors[]",
+                "tool.driver.ruleDescriptors[]",
+                "conversion.tool.driver.ruleDescriptors[]",
+                "tool.extensions[].ruleDescriptors",
+                "conversion.tool.extensions[].ruleDescriptors[]"
+            };
 
-        private static void UpdateToolReportingDescriptorPropertyTypes(JObject tool)
-        {
-            // Access and modify tool.driver
-            if (tool["driver"] is JObject driver)
-            {
-                UpdateToolComponentReportingDescriptorPropertyTypes(driver);
-            }
+            ApplyActionOnNodeIfPathExists(reportingDescriptorPathsToUpdate, run, UpdateReportingDescriptorPropertyTypes);
 
-            // Access and modify each item in tool.extensions
-            if (tool["extensions"] is JArray extensions)
-            {
-                foreach (JObject toolComponent in extensions)
-                {
-                    UpdateToolComponentReportingDescriptorPropertyTypes(toolComponent);
-                }
-            }
         }
 
-        private static void UpdateToolComponentReportingDescriptorPropertyTypes(JObject toolComponent)
+        private static void UpdateReportingDescriptorPropertyTypes(JObject reportingDescriptor)
         {
-            // Access and modify toolComponent.notificationDescriptors
-            if (toolComponent["notificationDescriptors"] is JArray notificationDescriptors)
+            if (reportingDescriptor["name"] is JObject message && message["text"] is JToken text)
             {
-                UpdateReportingDescriptorPropertyTypes(notificationDescriptors);
+                reportingDescriptor["name"] = text;
             }
 
-            // Access and modify toolComponent.ruleDescriptors
-            if (toolComponent["ruleDescriptors"] is JArray ruleDescriptors)
+            if (reportingDescriptor["shortDescription"] is JObject message2)
             {
-                UpdateReportingDescriptorPropertyTypes(ruleDescriptors);
-            }
-        }
-
-        private static void UpdateReportingDescriptorPropertyTypes(JArray reportingDescriptors)
-        {
-            foreach (JObject reportingDescriptor in reportingDescriptors)
```

Comment:
>foreach [](start = 12, length = 7)

the parser method will return all objects so it will take care of arrays as well as jobjects so this iteration in not needed.